clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -293,7 +323,7 @@ RValue CIRGenFunction::buildBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
     case Builtin::BI__builtin_copysignf16:
     case Builtin::BI__builtin_copysignl:
     case Builtin::BI__builtin_copysignf128:
-      llvm_unreachable("NYI");
+      return buildBinaryFPBuiltin<mlir::cir::CopysignOp>(*this, *E);
```

Comment:
If 16-bit fp is used seems like we're going to generate code that will fail to verify? Should tests for those be added or do we need assertions here?

Reply:
Updated. fp16 is not yet included in `CIR_AnyFloat` and thus the new operations cannot accept them yet. Made them unreachable.